pkg/tfbridge/info_defaults.go

```diff
@@ -0,0 +1,180 @@
+// Copyright 2016-2023, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tfbridge
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+	"unicode"
+
+	shim "github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/tokens"
+
+	multierror "github.com/hashicorp/go-multierror"
+)
+
+// SetDefaults populates mappings such as the Resources mapping by computing sensible defaults in code. This is
+// especially useful for larger providers that have thousands of entries in Resources. Manually specified mappings
+// always take precedence: SetDefaults will never override an entry in the Resources map, only add new entries.
+func (info *ProviderInfo) SetDefaults(defaults ProviderInfoDefaults) error {
```

Comment:
ComputeDefaults was a good name, just picking something here so it can exist side by side for discussion.